frontend/src/libs/components/customer-order-list/customer-order-list.tsx

```diff
@@ -0,0 +1,80 @@
+import { useCallback } from 'react';
+
+import { getValidClassNames } from '~/libs/helpers/helpers.js';
+import { type OrderResponseDto } from '~/packages/orders/orders.js';
+
+import { CustomerOrderCard } from '../customer-order-card/customer-order-card.jsx';
+import { Pagination } from '../pagination/pagination.jsx';
+import { Spinner } from '../spinner/spinner.jsx';
+import styles from './styles.module.scss';
+
+type Properties = {
+  orders: OrderResponseDto[];
+  isLoading?: boolean;
+  emptyListMessage?: JSX.Element;
+  pageSize: number;
+  totalElements: number;
+  pageIndex: number;
+  changePageIndex: React.Dispatch<React.SetStateAction<number>>;
+  changePageSize: React.Dispatch<React.SetStateAction<number>>;
+};
+
+const CustomerOrderList: React.FC<Properties> = ({
+  orders,
+  isLoading,
+  pageIndex,
+  pageSize,
+  emptyListMessage,
+  totalElements,
+  changePageIndex,
+  changePageSize,
+}: Properties) => {
+  const pagesRange = Math.ceil(totalElements / pageSize);
+
+  const createCustomerOrderCards = (): JSX.Element[] => {
+    return orders.map((order) => (
+      <li key={order.id} className={styles.item}>
+        <CustomerOrderCard order={order} />
+      </li>
+    ));
+  };
```

Comment:
I do not really get how you will replace the cycle with CustomerOrderCard to the markup like this 
`{orders.map((order) => (
      <li key={order.id} className={styles.item}>
        <CustomerOrderCard order={order} />
      </li>
    ))}`?

Reply:
It can be like ` const CustomerOrderCardsList = (): JSX.Element => {
    return (
      <ul>
        {orders.map((order) => (
          <li key={order.id} className={styles.item}>
            <CustomerOrderCard order={order} />
          </li>
        ))}
      </ul>
    );
  };
`

Reply:
I guess it doesn't affect performance much, but it looks cleaner to me

Reply:
as for me, it's ok both ways, but I'd prefer to leave it in the markup, like 
{`orders.map((order) => )}`